server.R

```diff
@@ -197,30 +167,70 @@ function(input, output, session) {
+                   sqrt(tauGSD/100*1/(tauGSD/100 + (1-tauGSD/100)*value$psiGSD)), 
+                   sqrt(tauGSD/100*1/(tauGSD/100 + (1-tauGSD/100)*value$psiGSD)), 
+                   1), ncol=2)
+    # calculate power for fixed design and GSD (first stage and final stage)
+    power.fix = 1-pnorm(qnorm(1-input$alphaGSD), mean = mu.t0 , sd=1)
+    power.gsd.1 = 1-pnorm(crit[1], mean = mu.t0, sd=1)
+    power.gsd.2 = 1-pmvnorm(lower = c(-Inf, -Inf), upper = c(crit[1], crit[2]), 
+                            mean = c(mu.t0 , mu.t), 
+                            sigma = SIG)[1]
+    data.gsd = rbind(powerGSD(),
+                     c(tauGSD, power.fix*100, power.gsd.1*100, power.gsd.2*100))
+    data.gsd = data.gsd[order(data.gsd[,1]),]
+    data.gsd = data.gsd[!duplicated(data.gsd[,1]),]
+    return(data.gsd)
   })
 
-  # The following code works but produces a warning message I couldn't resolve
-  # Maybe someone else has an idea?
-  output$power_tableGSD <- DT::renderDataTable({ # GSD Table in Tab: Power Information
+  output$power_plotlyGSD <- renderPlotly({ # GSD Plot in Tab: Power Information
+    data.gsd = powerGSDinput()
+    data_plot <- data.frame(Proportion=rep(data.gsd[, 1], 3), Power=as.vector(data.gsd[, 2:4]),
+                            Type=rep(c("Available", "GSD (stage 1)", "GSD (overall)"), each=nrow(data.gsd)))
+    data_plot$Type <- factor(data_plot$Type, levels=c("Available", "GSD (stage 1)", "GSD (overall)"))
+    data_plot$Power <- round(data_plot$Power,2)
+    p <- ggplot(data_plot, aes(x = Proportion, y = Power, color = Type)) +
+      geom_line(linetype = 1, size = 1) +
+      xlab("Proportion (%) of data available") +
+      ylab("Power (%)") +
+      xlim(0,100) +
+      ylim(0,100) +
+      theme_bw() +
+      scale_color_manual(values = c("Available" = "#0460A9", "GSD (stage 1)" = "#EC9A1E", "GSD (overall)" = "#8D1F1B")) +
+      theme(text = element_text(size = 12, face = "bold"),
+            plot.title = element_text(colour = "black", size = 12, face = "bold", hjust = 0),
+            legend.title = element_blank(),
+            plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm"),
+            legend.text = element_text(colour = "black", size = 12, face = "bold")
+      )
+    ggplotly(p)
     
-    data.gsd = powerGSD()
-    data.gsd[, 2:4] <- sprintf("%.2f",  powerGSD()[, 2:4])
-    data.gsd = data.frame(data.gsd)
-    colnames(data.gsd) <- c("tau", "Fixed", "GSD Stage 1", "GSD overall")
-    datatable(data.gsd, options = list(dom = 't')) %>% formatStyle(
+  })
+  
+  output$power_tableGSD <- DT::renderDataTable({ # GSD Table in Tab: Power Information
+    data.gsd = powerGSDinput()
+    data.table = data.gsd[which(round(data.gsd[,1],0)%%10==0 | data.gsd[,1]==input$tauGSD | (data.gsd[,1]>=79 & round(data.gsd[,1],1)%%5==0)),]
+    data.table[, 2:4] <- sprintf("%.2f",  data.table[, 2:4])
+    data.table = data.frame(data.table)
+    colnames(data.table) <- c("tau", "Available", "GSD Stage 1", "GSD overall")

```

Comment:
It would be good to make the table more accessible. Here are some suggestions.

Change "tau" to "Proportion (%) of data available"

"Available" to "Fixed sample test" or "Fixed sample design"

Add a title to indicate this table displays power.